events/syscall/modify_container_entrypoint.go

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package syscall
+
+import (
+    "os"
+    "github.com/falcosecurity/event-generator/events"
+)
+
+var _ = events.Register(
+    ModifyContainerEntrypoint,
+    events.WithDisabled(), // this rules is not included in falco_rules.yaml (stable rules), so disable the action
+)
+
+func ModifyContainerEntrypoint(h events.Helper) error {
+    if h.InContainer() {
+        // Open /proc/self/fd/1 for writing
+        file, err := os.OpenFile("/proc/self/fd/1", os.O_WRONLY, 0644)
+        if err != nil {
+            h.Log().WithError(err).Error("Failed to open /proc/self/fd/1 for writing")
+            return err
+        }
+        defer file.Close()
+
+        // Write "written by event-generator" to /proc/self/fd/1
+        data := []byte("written by event-generator")
+        _, err = file.Write(data)
```

Comment:
>  So, if we are altering the container filesystem and then deleting the container, does it matter to delete that change in the file?

It does not matter in the specific case you mentioned, but since the user can also run the action in any container they want (think when we deploy the event generator in a K8s cluster), we should not make such an assumption when implementing an action. Even the `h.InContainer()` guard isn't enough since we don't know if it's running in an ephemeral container or not.

(consideration: I know this is a limitation, but it's also a way to prevent unsafe usage of this tool, which is already dangerous, potentially :innocent: )

Reply:
@leogr please take a look at it again. One possibility is to use `os.truncate` . WDYT?
```
`os.truncate`: truncate any file where
`file.truncate`: truncate file objects which have been created or opened for writing, as it's a method of the file object.
```

Reply:
What's the advantage of using truncate? We will still modify `/proc/self/fd/1` and leave the container tampered with, right?

Just an alternative idea: we may append some data to `/proc/self/fd/1` and the revert it. Not sure if it's a good idea, but it's the only that I can find right now.

Reply:
@leogr I made changes, and took this approach: 

- Open `/proc/self/fd/1` in WRONLY mode
- Get the current size of file
- Write data into it
- Truncate the file to its initial size to remove the written content


This event triggering rule successfully too 
![Screenshot from 2024-04-12 23-15-08](https://github.com/falcosecurity/event-generator/assets/117383853/c438db69-8f60-4128-b4ff-78f386af1211)


To check its only removing written content by event, I tried this on /tmp directory by creating `/tmp/test.go` inside container and `main.go` includes this event. When I run this event, its only removing written content by this event not previously written content,

> $ sudo docker run -it container sh
> /app # echo "FALCO FALCO FALCO" > /tmp/test.go
> /app # cat /tmp/test.go 
> FALCO FALCO FALCO
> /app # go run main.go
> Modification completed successfully.
> /app # cat /tmp/test.go 
> FALCO FALCO FALCO
> /app # 
>   

@leogr @FedeDP what you think about this approach?

Reply:
The latest approach SMGT. I've just to look deeper into it.

cc @loresuso wdyt?

Reply:
Yes, I looked into this and after a refresher on the vuln the last approach sounds good to me as well to reproduce and trigger the rule